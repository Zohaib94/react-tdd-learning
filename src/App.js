import React, { Component } from 'react'

class App extends Component {
  state = {
    gifts: []
  }

  addGift = () => {
    let { gifts } = this.state
    gifts.push({ id: gifts.length })
    this.setState({ gifts })
  }

  render() {
    let { gifts } = this.state;

    return (
      <div>
        <h2>This is a testing application</h2>
        <button className='add-gift' onClick={this.addGift}>
          Add Gift
        </button>
        <div className='gift-list'>
          {
            gifts.map((gift) => (
              <div key={gift.id}>
                Gift #{gift.id}
              </div>
            ))
          }
        </div>
      </div>
    )
  }
}

export default App;
