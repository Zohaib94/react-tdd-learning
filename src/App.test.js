import React from 'react'
import { shallow } from 'enzyme'
import App from './App'

const app = shallow(<App/>);

it('is expected that component will render correctly', () => {
  expect(app).toMatchSnapshot();
})

it('is expected that component will initialize with empty array', () => {
  expect(app.state().gifts).toEqual([])
})

it('gift will be added on clicking the `Add Gift` button', () => {
  const previousGiftStateLength = app.state().gifts.length
  app.find('.add-gift').simulate('click')
  const currentGiftStateLength = app.state().gifts.length

  expect(currentGiftStateLength).toBeGreaterThan(previousGiftStateLength)
})

it('expects that a new gift is rendered after clicking `Add Gift Button`', () => {
  const previousGiftListLength = app.find('.gift-list').children().length
  app.find('.add-gift').simulate('click')
  const currentGiftListLength = app.find('.gift-list').children().length

  expect(currentGiftListLength).toBeGreaterThan(previousGiftListLength)
})
