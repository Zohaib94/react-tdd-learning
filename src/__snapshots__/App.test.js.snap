// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`is expected that component will render correctly 1`] = `
<div>
  <h2>
    This is a testing application
  </h2>
  <button
    className="add-gift"
    onClick={[Function]}
  >
    Add Gift
  </button>
  <div
    className="gift-list"
  />
</div>
`;
